### PR TITLE
Roadmap: Add Tetragon and remove GSoD

### PR DESCRIPTION
**Pull request by xmulligan**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

Fixes: #issue-number

```release-note
Added Tetragon to the roadmap
```

### CODE REVIEW

File: Documentation/community/roadmap.rst [hunk 1]
```diff
@@ -146,14 +147,16 @@ CI Test Improvements
...
+
+Tetragon provides security observability through the JSON events and the Tetragon
+CLI for things like process execution, file access, network observability, and
+privileged execution.
 
-Documentation Improvements
-~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-We would like to make this Cilium documentation :ref:`easier to
-navigate<season_of_docs>`, with more example configurations and links to
```
Comment (joestringer) [inline Documentation/community/roadmap.rst:155]: Should this PR also remove `Documentation/contributing/development/season_of_docs.rst` and the reference to season of docs from `Documentation/contributing/development/index.rst` ?

Comment (xmulligan) [reply Documentation/community/roadmap.rst:155]: I was going to remove them in a separate PR once the program is done in December. I'm guessing the graduation application will be in before then, but I'm not sure if it makes sense to have this in here for that

Comment (joestringer) [reply Documentation/community/roadmap.rst:155]: That works for me, just wanted to make sure we weren't forgetting to clean up some links. :ok_hand:

**Review by joestringer — APPROVED** [review]

File: Documentation/community/roadmap.rst [hunk 2]
```diff
@@ -33,16 +33,17 @@ Major Feature Status
 +-----------------------------+------------------------------------------------------------+
 | Service Mesh                | Stable (:ref:`Roadmap Details<rm-cilium-service-mesh>`)    |
 +-----------------------------+------------------------------------------------------------+
-
+| Tetragon security           | Beta                                                      |
```
Comment (borkmann) [inline Documentation/community/roadmap.rst:36]: this failed CI with malformed table:

```
3:50:46 PM: Building documentation (html)...
3:51:31 PM: Please fix the documentation warnings below
3:51:31 PM: /opt/build/repo/Documentation/community/roadmap.rst:17: ERROR: Malformed table.
3:51:31 PM: +-----------------------------+------------------------------------------------------------+
3:51:31 PM: | eBPF Networking             | Stable (:ref:`Roadmap Details<rm-advanced-networking>`)    |
3:51:31 PM: ++----------------------------+------------------------------------------------------------+
3:51:31 PM: || Kubernetes CNI             | Stable                                                     |
3:51:31 PM: ++----------------------------+------------------------------------------------------------+
3:51:31 PM: || Load Balancing             | Stable                                                     |
3:51:31 PM: ++----------------------------+------------------------------------------------------------+
3:51:31 PM: || Network Policy             | Stable                                                     |
3:51:31 PM: ++----------------------------+------------------------------------------------------------+
3:51:31 PM: || Kube-proxy Replacement     | Stable                                                     |
3:51:31 PM: ++----------------------------+------------------------------------------------------------+
3:51:31 PM: || Egress Gateway             | Stable                                                     |
3:51:31 PM: ++----------------------------+------------------------------------------------------------+
3:51:31 PM: | Multi-Cluster (ClusterMesh) | Stable (:ref:`Roadmap Details<rm-clustermesh>`)            |
3:51:31 PM: +-----------------------------+------------------------------------------------------------+
3:51:31 PM: | Hubble Observability        | Stable (:ref:`Roadmap Details<rm-hubble-observability>`)   |
3:51:31 PM: +-----------------------------+------------------------------------------------------------+
3:51:31 PM: | Service Mesh                | Stable (:ref:`Roadmap Details<rm-cilium-service-mesh>`)    |
3:51:31 PM: +-----------------------------+------------------------------------------------------------+
3:51:31 PM: | Tetragon security           | Beta                                                      |
3:51:31 PM: +-----------------------------+------------------------------------------------------------+
3:51:31 PM: /opt/build/repo/Documentation/community/roadmap.rst:38: WARNING: Blank line required after table.
3:51:31 PM: make: *** [Makefile:115: html-netlify] Error 1
3:51:31 PM: ​
3:51:31 PM: ────────────────────────────────────────────────────────────────
3:51:31 PM:   "build.command" failed                                        
3:51:31 PM: ────────────────────────────────────────────────────────────────
3:51:31 PM: ​
3:51:31 PM:   Error message
3:51:31 PM:   Command failed with exit code 2: make html-netlify (https://ntl.fyi/exit-code-2)
```

Comment (joestringer) [reply Documentation/community/roadmap.rst:36]: :disappointed: I could have sworn I checked and the Documentation target was marked as "passed". But I can see clearly by clicking above that it's not passing. I'll revert it, then @xmulligan can try again. Maybe leave the PR as draft until you have confirmed that the CI is passing, then mark it as ready for review.

**Comment by joestringer** [comment]
@xmulligan I ended up reverting due to the Documentation tests breakage, please resubmit :pray:

**Comment by xmulligan** [comment]
resubmitted in #21372 . I had the same problem again and it seems the issue was from how the online Github editor was set up

**Pull request merged**